plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -14,14 +14,61 @@
  * limitations under the License.
  */
 
-import React, { ChangeEvent, useEffect, useState } from 'react';
+import React, { ChangeEvent, useEffect, KeyboardEvent, useState } from 'react';
 import { useDebounce } from 'react-use';
 import { InputBase, InputAdornment, IconButton } from '@material-ui/core';
 import SearchIcon from '@material-ui/icons/Search';
 import ClearButton from '@material-ui/icons/Clear';
 
 import { useSearch } from '../SearchContext';
 
+type PresenterProps = {
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleClear?: () => void;
+  onKeyDown?: (e: KeyboardEvent<HTMLInputElement>) => void;
+  className?: string;
+  placeholder?: string;
+};
+
+export const SearchBarPresenter = ({
+  value,
+  onChange,
+  onKeyDown,
+  handleClear,
+  className,
```

Comment:
You don't need to change this, just mentioning that one pattern MUI uses in a couple of their components, is to have a `classes={{ ThingOne: string, ThingOne: string }}` pattern where you can apply class names to the different composed parts inside a thing. Maybe we'll be fine for now with just the edit box being styled because it's a simple structure in here, I just thought I'd mention it as an interesting aside for the future when you make larger components

Reply:
Another option is to have an `inputProps: InputBaseProps` but that is kinda leaky IMO so...

Reply:
That's a good call! In this instance, `className` was already implemented this way, so I didn't touch that API. For now, there is only one component AFAIK (the InputBase), so I think it's fine as is. Seems like something we could find useful in the future though!